### PR TITLE
Remove remaining occurances of "include_type_name=true" in docs

### PR DESCRIPTION
**Pull request by cbuescher**
This removes all the remaining uses of the "include_type_name" parameter
in the asciidocs that were temporary introduced in #37285. The only remaining
instances are in `removal_of_types.asciidoc` which warants a separate treatment
in a follow up.

### CODE REVIEW

File: docs/reference/mapping/types/object.asciidoc [hunk 1]
```diff
@@ -66,9 +64,8 @@ PUT my_index?include_type_name=true
 }
 --------------------------------------------------
 // CONSOLE
-<1> The mapping type is a type of object, and has a `properties` field.
```
Comment (jtibshirani) [inline docs/reference/mapping/types/object.asciidoc]: Same thought here as in the `properties` documentation.

File: docs/reference/mapping/params/properties.asciidoc [hunk 2]
```diff
@@ -58,10 +56,9 @@ PUT my_index/_doc/1 <4>
 }
 --------------------------------------------------
 // CONSOLE
-<1> Properties under the `_doc` mapping type.
```
Comment (jtibshirani) [inline docs/reference/mapping/params/properties.asciidoc]: Is this explanation maybe still useful to keep, without mention of a 'mapping type' (something like "Properties in the top-level mappings definition.")?

**Review by jtibshirani — APPROVED** [review]
I read over the changes and they look good to me! I left a small comment for your consideration, but am approving now as is it a minor point.

File: docs/reference/mapping/types/object.asciidoc (hunk 1 above)
Comment (cbuescher) [reply docs/reference/mapping/types/object.asciidoc]: I changes this.

File: docs/reference/mapping/params/properties.asciidoc (hunk 2 above)
Comment (cbuescher) [reply docs/reference/mapping/params/properties.asciidoc]: I changes this.

**Comment by cbuescher** [comment]
@elasticmachine run the gradle build tests 1

**Comment by cbuescher** [comment]
@elasticmachine run elasticsearch-ci/1

**Pull request merged**